### PR TITLE
AW-issue #6

### LINKED ISSUE #6: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [x] When I add a new image, there is a form field to add tags.
- [x] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [x] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by angelawang24**
Added tags to the image model
Allow users to save tags with new image

Closes #6

### CODE REVIEW

**Comment by angelawang24** [comment]
[Ropes issue #6](https://trello.com/c/8obXg8kx/1709-ropes-issue-6)

File: db/migrate/20190117172433_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This migration comes from acts_as_taggable_on_engine (originally 1)
+if ActiveRecord.gem_version >= Gem::Version.new('5.0')
+  class ActsAsTaggableOnMigration < ActiveRecord::Migration[4.2]; end
+else
+  class ActsAsTaggableOnMigration < ActiveRecord::Migration; end
+end
+ActsAsTaggableOnMigration.class_eval do
+  def self.up
+    create_table :tags do |t|
+      t.string :name
```
Comment (grahamotte) [inline db/migrate/20190117172433_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb]: It is a good habit to make these `null: false`, even if the generator doesn't be default :)

File: test/fixtures/images.yml [hunk 2]
```diff
@@ -2,9 +2,16 @@
 
 valid_img:
   url: 'https://cdn.pixabay.com/photo/2016/02/19/15/46/dog-1210559__340.jpg'
+  tags: 'dog, animal'
```
Comment (grahamotte) [inline test/fixtures/images.yml]: We mostly don't use fixtures in our apps and I think is a decent example of one of the reasons why. If you want to set some default, then you have to add it every fixture. Additionally, making changes in this way can break many tests. 

It's not that big of a deal for this project but something to keep in mind.

File: test/models/image_test.rb [hunk 3]
```diff
@@ -19,4 +19,15 @@ def test_invalid_img
     assert_equal "can't be blank", blank_url.errors.messages[:url].second
     assert_equal 'is not a valid image url', invalid_jpg_url.errors.messages[:url].first
   end
+
+  def test_tags_saved_to_img
```
Comment (grahamotte) [inline test/models/image_test.rb]: It is probably worth having a test to add new tags to an image and a test to remove tags.

**Pull request merged**